src/promptflow/promptflow/_sdk/_serving/app.py

```diff
@@ -105,6 +110,14 @@ def init_swagger(self):
         self.response_fields_to_remove = get_output_fields_to_remove(self.flow, logger)
         self.swagger = generate_swagger(self.flow, self.sample, self.response_fields_to_remove)
 
+    def set_default_environment_variables(self, default_environment_variables: Dict[str, str] = None):
+        if default_environment_variables is None:
+            return
+        for key, value in default_environment_variables.items():
+            if key not in os.environ:
+                os.environ[key] = value
+                self.environment_variables[key] = value
```

Comment:
same as the comment for self.environment_variables, If we don't need to keep it as a class field, no need to update it here.